include/oneapi/dpl/pstl/glue_algorithm_ranges_impl.h

```diff
@@ -314,6 +314,245 @@ struct __search_n_fn
 
 inline constexpr __internal::__search_n_fn search_n;
 
+namespace __internal
+{
+struct __count_if_fn
+{
+    template<typename _ExecutionPolicy, std::ranges::random_access_range _R, typename _Proj = std::identity,
+             std::indirect_unary_predicate<std::projected<std::ranges::iterator_t<_R>, _Proj>> _Pred>
+    requires oneapi::dpl::is_execution_policy_v<std::remove_cvref_t<_ExecutionPolicy>> && std::ranges::sized_range<_R>
+
+    std::ranges::range_difference_t<_R>
+    operator()(_ExecutionPolicy&& __exec, _R&& __r, _Pred __pred, _Proj __proj = {}) const
+    {
+        const auto __dispatch_tag = oneapi::dpl::__ranges::__select_backend(__exec);
+        return oneapi::dpl::__internal::__ranges::__pattern_count_if(__dispatch_tag,
+            std::forward<_ExecutionPolicy>(__exec), std::forward<_R>(__r), __pred, __proj);
+    }
+}; //__count_if_fn
+}  //__internal
+
+inline constexpr __internal::__count_if_fn count_if;
+
+namespace __internal
+{
+struct __count_fn
+{
+    template<typename _ExecutionPolicy, std::ranges::random_access_range _R, typename _Proj = std::identity,
+             typename _T>
+    requires oneapi::dpl::is_execution_policy_v<std::remove_cvref_t<_ExecutionPolicy>> && std::ranges::sized_range<_R>
+        && std::indirect_binary_predicate<std::ranges::equal_to, std::projected<std::ranges::iterator_t<_R>, _Proj>,
+        const _T*>
+    std::ranges::range_difference_t<_R>
+    operator()(_ExecutionPolicy&& __exec, _R&& __r, const _T& __value, _Proj __proj = {}) const
+    {
+        auto __pred = [__value](auto&& __val) { return std::ranges::equal_to{}(__val, __value);};
```

Comment:
It might be dangerous to capture by reference because we can feed a device kernel with that predicate. 
Also, I was not find another example in MVP1.